include/boolinq/boolinq.h

```diff
@@ -32,6 +32,7 @@ namespace boolinq {
     //
 
     struct LinqEndException {};
+    struct LinqNonMutableException {};
```

Comment:
Why can't utilize the same `LinqEndException ` exception for this mutable case?

Reply:
You can utilize the same exception. You can remove the LinqNonMutableException completely. 
But for Debugging and using the Lib I think it is cleaner to have a dedicated Exception.